app/models/spree/subscription.rb

```diff
@@ -127,6 +127,24 @@ def send_prior_notification
       end
     end
 
+    def cancel_stripe_subscription
+      begin
+        stripe_subscription = Stripe::Subscription.retrieve(self.stripe_subscription_id)
+        
+        if stripe_subscription
+          stripe_subscription.cancel
+        else
+          error_message = "Stripe subscription not found."
```

Comment:
just raise exception here like 
raise "stripe subsription not found for id=" + subscription id, 

it will be caught naturally by parent rescue where we are doing loggings.